library/socket/tcpsocket/recv_nonblock_spec.rb

```diff
@@ -27,6 +27,20 @@
     @socket.recv_nonblock(50).should == "TCPSocket#recv_nonblock"
   end
 
+  it 'writes the read to a buffer from the socket' do
+    @socket = TCPSocket.new @hostname, @server.port
+    @socket.write "TCPSocket#recv_nonblock"
+
+    # Wait for the server to echo. This spec is testing the return
+    # value, not the non-blocking behavior.
+    #
+    # TODO: Figure out a good way to test non-blocking.
+    IO.select([@socket])
+    buffer = "".b
+    @socket.recv_nonblock(50, buffer)
```

Comment:
It seems those methods need the flags as the second argument, so an extra `, 0`, for both cases.

Reply:
are you sure? According to the docs, the second arg can be either buffer or flags.

Reply:
you were right, fixed!

Reply:
The CI is still failing, this line wasn't changed it seems.

Reply:
https://docs.ruby-lang.org/en/3.0.0/BasicSocket.html#method-i-recv_nonblock
They are all positional, so it seems flags are needed.

Reply:
yup, that extra test for tcp socket failed. fixed now.